pycbc/psd/variation.py

```diff
@@ -210,10 +210,197 @@ def find_trigger_value(psd_var, idx, start, sample_rate):
+    upper frequency and an estimated signal response so that the variance
+    will be 1 when the filter is applied to white noise.
+
+    Within the PyCBC Live search this filter needs to be recreated every time
+    the estimated psd is updated and needs to be unique for each detector.
+
+    Parameters
+    ----------
+    psd_estimated : pycbc.frequencyseries
+        The estimated psd currently being used by the PyCBC Live search for
+        which we want to measure the psd variation of.
+    psd_duration : float
+        The duration of the estimation of the psd, in seconds.
+    sample_rate : int
+        The sample rate of the strain data being search over.
+    low_freq : int (default = 20)
+        The lower frequency to apply in the bandpass filter.
+    high_freq : int (default = 480)
+        The upper frequency to apply in the bandpass filter.
+
+    Returns
+    -------
+    full_filt : numpy.ndarray
+        The complete filter to be convolved with the strain data to
+        find the psd variation value.
+
+    """
+
+    # Create a bandpass filter between low_freq and high_freq once
+    filt = sig.firwin(4 * sample_rate,
+                      [low_freq, high_freq],
+                      pass_zero=False,
+                      window='hann',
+                      nyq=sample_rate / 2)
+    filt.resize(int(psd_duration * sample_rate), refcheck=False)
+
+    # Fourier transform the filter and take the absolute value to get
+    #  rid of the phase.
+    filt = abs(rfft(filt))
+
+    # Extract the psd frequencies to create a representative filter.
+    freqs = numpy.array(psd_estimated.sample_frequencies, dtype=numpy.float32)
+    plong = psd_estimated.numpy()
+
+    # Create the filter - bandpass, which weight by f^-7/6,
+    # and whiten. The normalization is chosen so that the variance
+    # will be one if this filter is applied to white noise which
+    # already has a variance of one.
+    fweight = freqs ** (-7./6.) * filt / numpy.sqrt(plong)
+    fweight[0] = 0.
+    norm = (sum(abs(fweight) ** 2) / (len(fweight) - 1.)) ** -0.5
+    fweight *= norm
+    fwhiten = numpy.sqrt(2. / sample_rate) / numpy.sqrt(plong)
+    fwhiten[0] = 0.
+    full_filt = sig.hann(int(psd_duration * sample_rate)) * numpy.roll(
+        irfft(fwhiten * fweight), int(psd_duration / 2) * sample_rate)
+
+    return full_filt
+
+
+def live_calc_psd_variation(strain,
+                            full_filt,
+                            increment,
+                            data_trim=2.0,
+                            short_stride=0.25):
+    """
+    Calculate the psd variation in the PyCBC Live search.
+
+    The Live strain data is convolved with the filter to produce a timeseries
+    containing the PSD variation values for each sample. This mean square of
+    the timeseries is then taken over the short_stride to remove the effects of
+    short duration glitches and further outliers from the mean are replaced
```

Comment:
This doesn't quite make sense.  'This mean square' ?  And does taking the mean square by itself remove effects of short duration glitches ?  Reading the doc for the existing mean square function seems to say it is used to identify and remove outliers due to glitches.

Reply:
'This mean square' -> 'The mean square'.

Yeah, the mean square identifies the outliers which are then removed and replaced. The mean square itself doesn't remove any effects of short duration glitches. I will reword this docstring.